sql/core/src/main/scala/org/apache/spark/sql/Dsl.scala

```diff
@@ -177,6 +181,23 @@ object Dsl {
     cols.toList.toSeq
   }
 
+  /**
+   * This is a private API for Python
+   * TODO: move this to a private package
+   */
+  def pythonUDF(
```

Comment:
we already have two functions here - why don't we take this chance to move both into a separate class?